### ISSUE TITLE
Quoting issue on build >= 420. Postgresql only. With $query->has|whereHas on attachMany relation.

### ISSUE THREAD

**Issue by joe-conigliaro** [octobercms/october#3011]
Update:
I have tested this on mysql & sqlite. It is a postgresql only issue.
This affects has, whereHas (probably all has queries) on attachOne/attachMany relationships.

##### Expected behavior
`$query->has('images')`
Should limit the query to results which have attachments on the images relation

##### Actual behavior
There is a quoting issue:
`... where exists (select * from "system_files" where "system_files"."attachment_id" = "CAST(table_name"."id" as "TEXT)" and "system_files"."attachment_type" = ...`

As you can see above the quoting is all messed up with the CAST. So far I havent been able to find out where this issue is originating from, it could be a laravel bug?

##### Reproduce steps
- Fresh install of octobercms build 420
- Use PostgreSQL DB Backend
- Must have a model with image attachment to test the query
- Try to limit the query with `$query->has('images')`

##### October build
Build >= 420

**Comment by LukeTowers** [comment]
@joe-conigliaro it would be a Laravel issue. Check the Laravel Database Postgresql Grammar for `has()` to dig deeper.

**Comment by LukeTowers** [comment]
Or do you have a typo in your model that you are running this on? Try posting the full code of the model you're trying to use this query with.

**Comment by joe-conigliaro** [comment]
Thanks @LukeTowers, I have been poking through the grammars and other files, but haven't been able to pinpoint the cause yet.

However the issue occurs on ANY model for example:

` $query = \Rainlab\Blog\Models\Post::has('featured_images');`
`$query->get();`

**Comment by LukeTowers** [comment]
Keep me updated please, good luck!

**Comment by joe-conigliaro** [comment]
I never found the root cause of this issue. For now I have modified like 68 of October\Rain\Database\Relations\AttachOneOrMany.php from: 
`$query = $query->select($columns)->whereColumn($this->getExistenceCompareKey(), '=', $key);` 
to: 
`$query = $query->select($columns)->whereColumn($this->getExistenceCompareKey(), '=', \DB::raw($key));`

Since this is a cast it maybe should be a raw expression? In either case there is a deeper problem :( but at least I could work around it for now

**Comment by LukeTowers** [comment]
Are you able to test this on MySQL and SQLlite and then provide reproduction instructions?

**Comment by joe-conigliaro** [comment]
Sure no problem, I will do the tests and provide instructions and get back. I do believe this is a postgresql only problem though as no one else has reported the issue.

**Comment by LukeTowers** [comment]
Thanks, that'll be helpful

**Comment by joe-conigliaro** [comment]
I have created a simple plugin to demonstrate the problem, simply

- download plugin: [kastro-testhas.zip](https://github.com/octobercms/october/files/1291462/kastro-testhas.zip)
- extract the plugin to the plugin directory of a fresh install of octobercms.
- setup/seed plugin: artisan october:up
- run console command to demonstrate bug: artisan testhasattachment:testhasquery

The plugin has a test model with an id field and a 'images' attachMany relation. It will automatically seed the database with a test row with attachment on 'october:up'. Then the console command 'testhasattachment:testhasquery' will attempt to run the offending query.

I have already tested on mysql and the problem does not exist.
